cdk_framework/EKS/lib/resource_constructs/general_constructs/test-case-resource-deployment-construct.ts

```diff
@@ -0,0 +1,79 @@
+import { Construct } from 'constructs';
+import { Cluster, FargateCluster } from 'aws-cdk-lib/aws-eks';
+import { NamespaceConstruct } from '../specific_constructs/namespace-construct';
+import { GeneralSampleAppDeploymentConstruct, GeneralSampleAppDeploymentConstructProps } from './general-sample-app-deployment-construct';
+import { GeneralCollectorDeploymentConstruct, GeneralCollectorDeploymentConstructProps } from './general-collector-deployment-construct';
+
+export class TestCaseResourceDeploymentConstruct extends Construct {
+    name: string
+    namespace: Construct
+
+    constructor(scope: Construct, id: string, props: TestCaseResourceDeploymentConstructProps){
+        super(scope, id);
+
+        // namespace deployment
+        const namespaceName = `collector-namespace`
+        const collectorNamespaceConstruct = new NamespaceConstruct(this, 'collector-namespace-construct', {
+            cluster: props.cluster,
+            name: namespaceName
+        })
+
+        // used for when sample app is push mode
+        const deployServices = props.sampleAppMode === 'push'
```

Comment:
Isn't a service required for both push and pull sample apps?

Reply:
Ya you're right. It will be necessary for the validator to be able to access the metrics.